.github/workflows/check-for-workflow-run.js

```diff
@@ -15,9 +15,10 @@ module.exports = async ({github, context, core, workflow_id, sha, ...config}) =>
   // run for this SHA we see.
   const runFilterDesc = sha ? `${workflow_id} on ${sha}` : workflow_id
   for (const run of result.data.workflow_runs) {
-    if (!sha || run.head_sha == sha) {
-      if (run.conclusion != "success") {
+    if ((!sha || run.head_sha === sha) && run.conclusion !== "in_progress") {
+      if (run.conclusion !== "success") {
         core.setFailed(`Last run of ${runFilterDesc} did not succeed: ${run.html_url}`)
+        return
```

Comment:
Can you pull that out?

Reply:
I will, but I wonder whether it's worth the extra build trip ^,^